elytron/src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java

```diff
@@ -710,18 +707,22 @@ public void registerOperations(ManagementResourceRegistration resourceRegistrati
                 resourceRegistration.registerOperationHandler(new SimpleOperationDefinitionBuilder(ElytronDescriptionConstants.RELOAD_CERTIFICATE_REVOCATION_LIST, getResourceDescriptionResolver())
                             .setRuntimeOnly()
                             .build()
-                        , new ReloadCertificateRevocationList());
-            }
-
-            class ReloadCertificateRevocationList extends ElytronRuntimeOnlyHandler {
-
-                private ReloadCertificateRevocationList() {
-                }
-
-                @Override
-                protected void executeRuntimeStep(OperationContext context, ModelNode operation) throws OperationFailedException {
-                    reloadCrl.compareAndSet(false, true);
-                }
+                        , new ElytronRuntimeOnlyHandler() {
+                            @Override
+                            protected void executeRuntimeStep(OperationContext context, ModelNode operation) throws OperationFailedException {
+                                ServiceName serviceName = TRUST_MANAGER_RUNTIME_CAPABILITY.fromBaseCapability(context.getCurrentAddressValue()).getCapabilityServiceName();
+                                ServiceController<TrustManager> serviceContainer = getRequiredService(context.getServiceRegistry(true), serviceName, TrustManager.class);
+                                State serviceState;
+                                if ((serviceState = serviceContainer.getState()) != State.UP) {
+                                    throw ROOT_LOGGER.requiredServiceNotUp(serviceName, serviceState);
+                                }
+                                TrustManager trustManager = serviceContainer.getValue();
+                                if (! (trustManager instanceof ReloadableX509ExtendedTrustManager)) {
+                                    throw ROOT_LOGGER.unableToReloadCRLNotReloadable();
```

Comment:
I think this should throw OFE, not RuntimeException. A RuntimeException means some sort of server fault, and it will lead to an error in the logs. The user trying to perform an operation on a resource not configured to allow that operation is a user mistake, and that's an OFE.

My assumption here is it's valid for trustManager to be something other than ReloadableX509ExtendedTrustManager; i.e. that this check isn't doing something like an "assert trustManager instanceof ReloadableX509ExtendedTrustManager;" would. If it's like an assert then don't throw OFE.